swagger/swagger-generator/generator-core/src/main/java/org/apache/servicecomb/swagger/generator/core/utils/ParamUtils.java

```diff
@@ -113,12 +111,16 @@ public static BodyParameter createBodyParameter(Swagger swagger, String paramNam
   }
 
   public static void addDefinitions(Swagger swagger, Type paramType) {
+    if (swagger.getDefinitions() == null) {
+      ModelResolverExt.refreshClassNameMap();
```

Comment:
this makes swagger generator mechanism from thread safe to not thread safe

Reply:
can i use  ThreadLocal fix?

Reply:
seems we can check this depend on swagger instance, no need to modify ModelResolverExt.
we can save a instance in swagger.vendorExtensions, note: key should not start with "x-"

Reply:
According to my understanding，if use swaggger we must get the type then store in vendorExtensions.  the way we get the type is very complex ,cause we must handle like list<T>, map<T>, nesting and so on . you can see my first commit '9a2b56a1b2d8535ee0d211ee990fe0957f028825'. i found the ModelResolverExt can help me get type later , and i take the way you see now.

Reply:
```java
  public static void addDefinitions(Swagger swagger, Type paramType) {
    Map<String, Model> models = ModelConverters.getInstance().readAll(paramType);
    if (null!=swagger.getDefinitions()) {
      for (Entry<String, Model> modelEntry : models.entrySet()) {
        Model tempModel = swagger.getDefinitions().get(modelEntry.getKey());
        if (null != tempModel && !tempModel.equals(modelEntry.getValue())) {
          throw new IllegalArgumentException("duplicate param model: " + modelEntry.getKey());
        }
      }
    }
    for (Map.Entry<String, Model> entry : models.entrySet()) {
      swagger.addDefinition(entry.getKey(), entry.getValue());
    }
  }
```
Seems the logic like above can works? It's in `org.apache.servicecomb.swagger.generator.core.utils.ParamUtils#addDefinitions` method. As long as all of the subclasses of `Model` implement `equals` properly, we can just check the definition map.

Reply:
ok, i refactory my code